### ISSUE TITLE
SpotRaop quits with exec_request:625 [0x7fac00d880]: response :  request failed

### ISSUE THREAD

**Issue by nano9g** [philippe44/SpotConnect#18]
I ran into this one while testing with the HomePod. I attempted to play a track by connecting from the Spotify devices menu, and a few seconds later after what seemed like mostly normal log messages, SpotRaop said `exec_request:625 [0x7fac00d880]: response :  request failed` and quit to the command line.

I don't know how to reproduce this because when I started SpotRaop back up and tried to connect again, it worked normally.

It does seem like this is something SpotRaop could try to handle more gracefully though. 🙂

<details>
<summary><strong>📃 Log</strong></summary>

```
$ ./spotconnect/spotraop-linux-aarch64-static -x ./spotraop-config.xml -d all=debug
[13:42:05.370361] main:1137 Starting spotraop version: v0.1.5 (Aug  6 2023 @ 16:57:26)

[13:42:05.378339] Start:906 Binding to [Main Network IP]
[13:42:05.378872] StartActiveRemote:788 DACP port: 33615
[13:42:05.493610] AddRaopDevice:528 [0xa56e10]: adding renderer (Kitchen-HomePod@[Main Network IP]) with mac AAAA-XXXXXXXX
[13:42:05.493937] raopcl_create:713 [0x7fac0094e0]: using ALAC coding
I BellHTTPServer.cpp:191: Server listening on port 0
[13:42:05.500539] AddRaopDevice:481 [0xa572c8]: AppleTV with valid authentication key [key]
[13:42:05.501050] AddRaopDevice:528 [0xa572c8]: adding renderer (Den-Apple-TV@[Main Network IP]) with mac AAAA-XXXXXXXX
[13:42:05.501533] raopcl_create:713 [0x7fac0141c0]: using ALAC coding
I BellHTTPServer.cpp:191: Server listening on port 0
[13:42:05.502473] AddRaopDevice:514 [0xa57780]: creating MAC
[13:42:05.502842] AddRaopDevice:528 [0xa57780]: adding renderer (DEVF6GT4QEDWK3TEFQNZ@[IOT Network IP]) with mac AAAA-XXXXXXXX
[13:42:05.503769] raopcl_create:713 [0x7fac01dec0]: using ALAC coding
I BellHTTPServer.cpp:191: Server listening on port 0
I spotify.cpp:366: Server using actual port 36063
I MDNSService.cpp:181: using built-in mDNS for Kitchen HomePod
I spotify.cpp:366: Server using actual port 46323
I MDNSService.cpp:181: using built-in mDNS for Den Apple TV
I spotify.cpp:366: Server using actual port 41375
I MDNSService.cpp:181: using built-in mDNS for Shop Cone+
[13:42:05.657234] AddRaopDevice:481 [0xa57c38]: AppleTV with valid authentication key [key]
[13:42:05.657377] AddRaopDevice:528 [0xa57c38]: adding renderer (Sunroom-Apple-TV@[Main Network IP]) with mac AAAA-XXXXXXXX
[13:42:05.657637] raopcl_create:713 [0x7fac027ed0]: using ALAC coding
I BellHTTPServer.cpp:191: Server listening on port 0
I spotify.cpp:366: Server using actual port 41209
I MDNSService.cpp:181: using built-in mDNS for Sunroom Apple TV
I spotify.cpp:388: Spotify client connected for Kitchen HomePod
D Session.cpp:63: Connecting with AP <ap-gue1.spotify.com:4070>
D PlainConnection.cpp:101: Connected to spotify server
I Session.cpp:39: Received APHello response
D Session.cpp:44: Received shannon keys
D Session.cpp:82: Authorization successful
I MercurySession.cpp:42: Received packet, command: 4
D TimeProvider.cpp:15: Time synced with spotify servers
I MercurySession.cpp:42: Received packet, command: 2
I MercurySession.cpp:42: Received packet, command: 118
D MercurySession.cpp:251: Executing Mercury Request, type SUB
I MercurySession.cpp:42: Received packet, command: 27
D spotify.cpp:423: keepAlive Kitchen HomePod
D MercurySession.cpp:153: Received country code US
I MercurySession.cpp:42: Received packet, command: 80
I MercurySession.cpp:42: Received packet, command: 31
I MercurySession.cpp:42: Received packet, command: 105
I MercurySession.cpp:42: Received packet, command: 181
I AccessKeyFetcher.cpp:99: Access token expired, fetching new one... 238
I MercurySession.cpp:42: Received packet, command: 74
I MercurySession.cpp:42: Received packet, command: 179
D MercurySession.cpp:174: Received mercury packet
D MercurySession.cpp:251: Executing Mercury Request, type SEND
D SpircHandler.cpp:59: Sent kMessageTypeHello!
I MercurySession.cpp:42: Received packet, command: 181
I MercurySession.cpp:42: Received packet, command: 181
D SpircHandler.cpp:67: Received subscription response
I MercurySession.cpp:42: Received packet, command: 178
D SpircHandler.cpp:124: Notify frame
D SpircHandler.cpp:67: Received subscription response
D SpircHandler.cpp:124: Notify frame
D MercurySession.cpp:174: Received mercury packet
I AccessKeyFetcher.cpp:114: Access token sucessfully fetched
I MercurySession.cpp:42: Received packet, command: 181
D SpircHandler.cpp:67: Received subscription response
D SpircHandler.cpp:124: Notify frame
I MercurySession.cpp:42: Received packet, command: 181
D SpircHandler.cpp:67: Received subscription response
D SpircHandler.cpp:170: Load frame 10!
D MercurySession.cpp:251: Executing Mercury Request, type SEND
I TrackPlayer.cpp:94: Resetting state
D MercurySession.cpp:251: Executing Mercury Request, type GET
I MercurySession.cpp:42: Received packet, command: 178
D MercurySession.cpp:174: Received mercury packet
I TrackQueue.cpp:158: Track name: Moto Verse
I TrackQueue.cpp:159: Track duration: 176069
D TrackQueue.cpp:161: trackInfo.restriction.size() = 1
D TrackQueue.cpp:212: File format: 2
D TrackQueue.cpp:212: File format: 1
D TrackQueue.cpp:212: File format: 0
I MercurySession.cpp:42: Received packet, command: 178
D MercurySession.cpp:174: Received mercury packet
I MercurySession.cpp:42: Received packet, command: 13
I TrackQueue.cpp:252: Got audio key
I TrackQueue.cpp:275: Received access key, fetching CDN URL...
I TrackQueue.cpp:301: Received CDN URL, https://audio-fa.scdn.co/audio/503cfd0bcd546647ef078e0c290f9339874c6d54?1691516544_HZDq-Qx6XgkfIQszx13pezeGDFOR6f-lwAld2swHvOg=
I TrackPlayer.cpp:167: Got track ID=503cfd0bcd546647ef078e0c290f9339874c6d54
I CDNAudioFile.cpp:43: Opening HTTP stream to https://audio-fa.scdn.co/audio/503cfd0bcd546647ef078e0c290f9339874c6d54?1691516544_HZDq-Qx6XgkfIQszx13pezeGDFOR6f-lwAld2swHvOg=
D MercurySession.cpp:251: Executing Mercury Request, type GET
I MercurySession.cpp:42: Received packet, command: 178
D MercurySession.cpp:174: Received mercury packet
I TrackQueue.cpp:158: Track name: Orb Two
I TrackQueue.cpp:159: Track duration: 184298
D TrackQueue.cpp:161: trackInfo.restriction.size() = 1
D TrackQueue.cpp:212: File format: 2
D TrackQueue.cpp:212: File format: 1
D TrackQueue.cpp:212: File format: 0
I CDNAudioFile.cpp:70: Header and footer bytes received
D MercurySession.cpp:251: Executing Mercury Request, type SEND
I spotify.cpp:235: new track will start at 0
[13:42:24.866597] shadowRequest:219 [0xa56e10]: spotify LOAD request
I MercurySession.cpp:42: Received packet, command: 13
I TrackQueue.cpp:252: Got audio key
I TrackQueue.cpp:275: Received access key, fetching CDN URL...
[13:42:24.970621] raopcl_connect:996 [0x7fac0094e0]: local interface [Main Network IP]
[13:42:24.971255] exec_request:614 [0x7fac00d880]: ----> : write ANNOUNCE rtsp://[Main Network IP]/2615159066 RTSP/1.0
Content-Type: application/sdp
Content-Length: 178
CSeq: 1
User-Agent: iTunes/7.6.2 (Windows; N;)
Client-Instance: 7d4257d25816cb73
DACP-ID: 1A2B3D4EA1B2C3D4
Active-Remote: 1405334755

v=0
o=iTunes 2615159066 0 IN IP4 [Main Network IP]
s=iTunes
c=IN IP4 [Main Network IP]
t=0 0
m=audio 0 RTP/AVP 96
a=rtpmap:96 AppleLossless
a=fmtp:96 352 0 16 40 10 14 2 255 0 0 44100

I MercurySession.cpp:42: Received packet, command: 178
D MercurySession.cpp:174: Received mercury packet
I TrackQueue.cpp:301: Received CDN URL, https://audio-fa.scdn.co/audio/53b2cb27ad071489f2498ce45c7d69fb800583c5?1691516545_aUomyCOqjYTYH3EHlFYrzNTzLmviuhkm9eLCaV9HByk=
D MercurySession.cpp:251: Executing Mercury Request, type GET
I MercurySession.cpp:42: Received packet, command: 178
D MercurySession.cpp:174: Received mercury packet
I TrackQueue.cpp:158: Track name: Morph
I TrackQueue.cpp:159: Track duration: 348644
D TrackQueue.cpp:161: trackInfo.restriction.size() = 1
D TrackQueue.cpp:212: File format: 2
D TrackQueue.cpp:212: File format: 1
D TrackQueue.cpp:212: File format: 0
I MercurySession.cpp:42: Received packet, command: 13
I TrackQueue.cpp:252: Got audio key
I TrackQueue.cpp:275: Received access key, fetching CDN URL...
I TrackQueue.cpp:301: Received CDN URL, https://audio-fa.scdn.co/audio/d8fc53a027322cffaaa37cf3a459daaef43d774f?1691516545_rtL_LFPctPVM7PZokBL9-XjQ96xwsUmwDrICxLeW4uQ=
[13:42:34.982234] exec_request:625 [0x7fac00d880]: response :  request failed
Segmentation fault
```

</details>

**Comment by philippe44** [comment]
It's not supposed to abort() on error there. I don't know yet, but that a crash

**Comment by philippe44** [comment]
That one should be fixed in 0.1.8
